File: gymcrm.web/src/modules/billing/types/insertPayment.ts
// Mirrors GymCRM.BillingAPI.Models.DTOs.InsertPayment 1:1.
export interface InsertPayment {
    subscriptionId: string;
    amount: number;
    method: number;
    status: number;
    paidAt?: string;
    externalReference?: string;
}

File: GymCRM.BillingAPI/Models/DTOs/InsertPayment.cs
using System.ComponentModel.DataAnnotations;

namespace GymCRM.BillingAPI.Models.DTOs;

public class InsertPayment
{
    [Required]
    public Guid SubscriptionId { get; set; }
    [Required]
    public decimal Amount { get; set; }
    [Required]
    public int Method { get; set; }
    [Required]
    public int Status { get; set; }
    public DateTime? PaidAt { get; set; }
    public string? ExternalReference { get; set; }
}
